scipy/stats/_distn_infrastructure.py

```diff
@@ -1668,7 +1666,10 @@ def _ppf_single(self, q, *args):
                 right = left
                 left *= factor
             # left is now such that cdf(left) < q
-        if not right:  # i.e. self.b = inf
+
+        if _b < np.inf:
```

Comment:
what is the reason for this change?

Reply:
Cleaned up `_ppf_single` to make its intent clearer.